src/reduxTypes.js

```diff
@@ -357,6 +357,17 @@ export type GlobalState = $ReadOnly<{|
   users: UsersState,
 |}>;
 
+// No substate should allow `undefined`; our use of AsyncStorage
+// depends on it. (This check will also complain on `null`, which I
+// don't think we'd have a problem with. We could try to write this
+// differently if we want to allow `null`.)
+type NonMaybeProperties<O: { ... }> = $ObjMap<O, <V>(V) => $NonMaybeType<V>>;
+type NonMaybeGlobalState = NonMaybeProperties<GlobalState>;
+// $FlowIgnore[incompatible-type]: Pretend `s` is a `GlobalState`.
+const s: GlobalState = 'nothing-to-see-here';
+// This cast will fail typechecking if GlobalState is wrong.
+(s: NonMaybeGlobalState);
```

Comment:
Here's a slightly cleaner way to write this -- avoids creating a bogus value, by making the `GlobalState` just the hypothetical parameter to a certain function:

```suggestion
// This function definition will fail typechecking if GlobalState is wrong.
(s: GlobalState): NonMaybeGlobalState => s; // eslint-disable-line no-unused-expressions
```